cms/static/js/spec/views/login_studio_spec.js

```diff
@@ -0,0 +1,30 @@
+define(['jquery', 'js/factories/login', 'common/js/spec_helpers/ajax_helpers', 'common/js/components/utils/view_utils'],
+function($, LoginFactory, AjaxHelpers, ViewUtils) {
+    'use strict';
+    describe("Studio Login Page", function() {
+        var submit_button;
+
+        beforeEach(function() {
+            loadFixtures('login.underscore');
+            LoginFactory("/home/");
+            submit_button = $('#submit');
+        });
+
+        it('It will disable the submit button once its clicked', function() {
+            spyOn(ViewUtils, 'redirect').andCallFake(function(){});
+            var requests = AjaxHelpers.requests(this);
+            expect(submit_button).not.toHaveClass('is-disabled');
+            submit_button.click();
+            AjaxHelpers.respondWithJson(requests, {'success': true});
+            expect(submit_button).toHaveClass('is-disabled');
```

Comment:
The reason why I think it's smarter to user `toHaveBeenCalledWith` is because we want to make sure that `is-disabled` was applied _and_ not removed. In the test below, for example, it's not just enough to test that the submit button doesn't have the class `is-disabled` — if you just made clicking the submit button do nothing, that test would still pass. For that test, you want to make sure that the class was applied and then removed.

So, the reason to use `toHaveBeenCalledWith` in this instance is that what we are testing is that the `is-disabled` class was added and removed, or added and not removed. You can keep the `toHaveClass` tests if you like, but it's essential still to have the `toHaveBeenCalledWith` tests as well.

Make sense?